tests/rptest/tests/datalake/schema_evolution_test.py

```diff
@@ -20,210 +29,560 @@
+                             **kwargs)
+        self.test_ctx = test_ctx
+        self.topic_name = "test"
+        self.table_name = f"redpanda.{self.topic_name}"
 
     def setUp(self):
+        # redpanda will be started by DatalakeServices
         pass
 
-    def _create_schema(self, subject: str, schema: str, schema_type="avro"):
-        rpk = RpkTool(self.redpanda)
-        with tempfile.NamedTemporaryFile(suffix=f".{schema_type}") as tf:
-            tf.write(bytes(schema, 'UTF-8'))
-            tf.flush()
-            rpk.create_schema(subject, tf.name)
+    def _select(self,
+                dl: DatalakeServices,
+                query_engine: QueryEngineType,
+                cols=list[str],
+                sort_by_offset: bool = True):
+        qe = dl.spark() if query_engine == QueryEngineType.SPARK else dl.trino(
+        )
+        query = f"select redpanda.offset, {', '.join(cols)} from {self.table_name}"
+        self.redpanda.logger.debug(f"QUERY: '{query}'")
+        out = qe.run_query_fetch_all(query)
+        if sort_by_offset:
+            out.sort(key=lambda r: r[0])
+        return out
+
+    def produce(
+        self,
+        dl: DatalakeServices,
+        schema: AvroSchema,
+        count: int,
+        context: TranslationContext,
+        should_translate: bool = True,
+    ):
+        producer = AvroProducer(
+            {
+                'bootstrap.servers': self.redpanda.brokers(),
+                'schema.registry.url': self.redpanda.schema_reg().split(",")[0]
+            },
+            default_value_schema=schema.load())
+
+        for i in range(count):
+            #TODO(oren): use something other than the index here?
+            record = schema.gen(context.total + i)
+            producer.produce(topic=self.topic_name, value=record)
+
+        producer.flush()
+        if should_translate:
+            dl.wait_for_translation(self.topic_name,
+                                    msg_count=context.total + count)
+            context.total = context.total + count
+            return
+
+        with expect_exception(TimeoutError, lambda _: True):
+            dl.wait_for_translation(self.topic_name,
+                                    msg_count=context.total + count,
+                                    timeout=10)
 
-    @cluster(num_nodes=4, log_allow_list=["Schema \d+ already exists"])
+    def check_table_schema(
+        self,
+        dl: DatalakeServices,
+        query_engine: QueryEngineType,
+        expected: AvroSchema,
+    ):
+        qe = dl.spark() if query_engine == QueryEngineType.SPARK else dl.trino(
+        )
+        table = qe.run_query_fetch_all(f"describe {self.table_name}")
+
+        if query_engine == QueryEngineType.SPARK:
+            table = [(t[0], t[1]) for t in table[1:-3]]
+        elif query_engine == QueryEngineType.TRINO:
+            table = [(t[0], t[1]) for t in table[1:]]
+
+        assert table == expected.table(query_engine), \
+            str(table)
```

Comment:
Can we add a little more color here about what was expected?

Reply:
Protobuf diff is close behind, i'll be sure to improve these assert messages in there.